generators/app/templates/src/redux/auth/actions.js

```diff
@@ -0,0 +1,67 @@
+import { NavigationActions } from 'react-navigation';
+
+import * as AuthService from '../../services/AuthService';
+import { stringArrayToObject } from '../../utils/reduxUtils';
+
+export const actions = stringArrayToObject(
+  ['LOGIN', 'LOGIN_SUCCESS', 'LOGIN_FAILURE', 'LOGOUT', 'AUTH_INIT'],
+  '@@AUTH'
+);
+
+const privateActionCreators = {
+  loginSuccess(authData) {
+    return {
+      type: actions.LOGIN_SUCCESS,
+      payload: { authData }
+    };
+  },
+  loginFailure(err) {
+    return {
+      type: actions.LOGIN_FAILURE,
+      payload: { err }
+    };
+  }
+};
+
+export const actionCreators = {
+  init(user) {
+    return {
+      type: actions.AUTH_INIT,
+      payload: { user }
+    };
+  },
+  login(authData) {
+    return async dispatch => {
+      // TODO: Call api here
+      authData = authData || {};
+      authData.sessionToken = 'token';
+
+      dispatch({ type: actions.LOGIN });
+      try {
+        await AuthService.setCurrentUser(authData);
+        await new Promise(resolve => setTimeout(resolve, 750));
```

Comment:
Mmm, can we extract this mock to the Service ? So API calls are transparent for this action